File: static/java/FormulaNumberField.java
// ==========================================
// MyAirtable static runtime.
// ==========================================

package myairtable;

/** Formula builder for number fields. */
public final class FormulaNumberField implements FormulaField {

  private final String fieldId;

  public FormulaNumberField(String fieldId) {
    this.fieldId = fieldId;
  }

  @Override
  public String getFieldId() {
    return fieldId;
  }

  /** Equal to value. Named {@code eq} rather than {@code equals} (§2.3.2). */
  public String eq(Number value) {
    return field() + "=" + value;
  }

  /** Not equal to value. Named {@code neq} for symmetry with {@code eq} (§2.3.2). */
  public String neq(Number value) {
    return field() + "!=" + value;
  }

  /** Greater than value. */
  public String greaterThan(Number value) {
    return field() + ">" + value;
  }

  /** Less than value. */
  public String lessThan(Number value) {
    return field() + "<" + value;
  }

  /** Greater than or equal to value. */
  public String greaterThanOrEquals(Number value) {
    return field() + ">=" + value;
  }

  /** Less than or equal to value. */
  public String lessThanOrEquals(Number value) {
    return field() + "<=" + value;
  }

  /** Between min and max, inclusive. */
  public String between(Number min, Number max) {
    return between(min, max, true);
  }

  /** Between min and max. */
  public String between(Number min, Number max, boolean inclusive) {
    return inclusive
        ? Formulas.and(greaterThanOrEquals(min), lessThanOrEquals(max))
        : Formulas.and(greaterThan(min), lessThan(max));
  }
}

File: src/myairtable/static/csharp/FormulaNumberField.cs
namespace MyAirtable;

/// <summary>Formula builder for number fields.</summary>
public sealed class FormulaNumberField : FormulaField
{
    public FormulaNumberField(string fieldId)
        : base(fieldId) { }

    /// <summary>Equal to value. Named <c>Eq</c> rather than <c>Equals</c>.</summary>
    public string Eq(double value) => Field() + "=" + Formulas.Num(value);

    /// <summary>Not equal to value. Named <c>Neq</c> for symmetry with <c>Eq</c>.</summary>
    public string Neq(double value) => Field() + "!=" + Formulas.Num(value);

    /// <summary>Greater than value.</summary>
    public string GreaterThan(double value) => Field() + ">" + Formulas.Num(value);

    /// <summary>Less than value.</summary>
    public string LessThan(double value) => Field() + "<" + Formulas.Num(value);

    /// <summary>Greater than or equal to value.</summary>
    public string GreaterThanOrEquals(double value) => Field() + ">=" + Formulas.Num(value);

    /// <summary>Less than or equal to value.</summary>
    public string LessThanOrEquals(double value) => Field() + "<=" + Formulas.Num(value);

    /// <summary>Between min and max.</summary>
    public string Between(double min, double max, bool inclusive = true) =>
        inclusive
            ? Formulas.And(GreaterThanOrEquals(min), LessThanOrEquals(max))
            : Formulas.And(GreaterThan(min), LessThan(max));
}
